packages/rollup-core/src/app/data/data-service.ts

```diff
@@ -481,19 +487,23 @@ export class DefaultDataService implements DataService {
       throw Error(msg)
     }
 
-    let totalCalldataBytes: number = 0
+    let totalL1TxBytes: number =
+      L1_ROLLUP_BATCH_TX_STATIC_OVERHEAD_BYTES +
+      L1_ROLLUP_BATCH_TX_STATIC_CALLDATA_BYTES
     let lastBlockNumber = -1
     for (const row of res) {
       const rowBytes: number = parseInt(row['calldata_bytes'], 10)
-      totalCalldataBytes += rowBytes
-      if (totalCalldataBytes > maxBatchCalldataBytes) {
+      totalL1TxBytes +=
+        roundToNearestMultipleOf32(rowBytes) +
+        L1_ROLLUP_BATCH_TX_BYTES_PER_L2_TX
```

Comment:
One idea that I had just reading this is you could potentially even try creating an Ethers `canonicalTxChain: Contract` object and literally calling [`canonicalTxChain.interface. encodeFunctionData(batch)`](https://docs.ethers.io/v5/api/utils/abi/interface/#Interface--encoding) to get the length of the ABI encode object. This is definitely less efficient but it at least does remove some human error.

That said I did double check the numbers so this should return the correct value, I believe

Reply:
I think this could be a good idea as a sanity check at the end, but I was a bit concerned about calling it every iteration on each tx. It's quite possible this concern is unfounded, though, and there would be minimal speed impact calling encode + sign hundreds of times in the worst case. 

I'm happy to change this or to profile it if you think we should, though!